navigator/backend/src/main/scala/com/digitalasset/navigator/UIBackend.scala

```diff
@@ -119,44 +128,41 @@ abstract class UIBackend extends LazyLogging with ApplicationInfoJsonSupport {
             } ~
               path("session"./) {
                 get {
-                  session match {
-                    case Some((_, session)) => complete(session)
-                    case None => complete(signIn)
+                  complete {
+                    session match {
+                      case Some((_, session)) => session
+                      case None => signIn()
+                    }
                   }
                 } ~
                   post {
                     entity(as[LoginRequest]) { request =>
-                      if (arguments.requirePassword) {
-                        request.maybePassword match {
-                          case None =>
-                            logger.error(
-                              s"Attempt to signin with user ${request.userId} without password")
-                            complete(invalidCredentials)
-                          case Some(password) =>
-                            config.users.get(request.userId) match {
-                              case None =>
-                                logger.error(
-                                  s"Attempt to signin with non-existent user ${request.userId}")
-                                complete(invalidCredentials)
-                              case Some(userConfig)
-                                  if userConfig.password == None || userConfig.password.contains(
-                                    password) =>
+                      config.users.get(request.userId) match {
+                        case None =>
+                          logger.error(
+                            s"Attempt to signin with non-existent user ${request.userId}")
+                          complete(signIn(Some(InvalidCredentials)))
+                        case Some(userConfig) =>
+                          import system.dispatcher
+                          Await.result(
```

Comment:
Instead of `Await.result` you can use the `onSuccess` directive.

Reply:
I didn't know about `akka-http`'s `onSuccess`, thanks for the tip!

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3138/commits/609adf670962eca159a75ff0e74773dc7930aca2